animation/animation.ino

```diff
@@ -0,0 +1,394 @@
+
+  if(is_button_pressed == true & last_buttonpress == true){
+  pos++;
+  
+  }
+  //
+  if(is_button_pressed == false & last_buttonpress == true) {
+    global_hue = random8();
+  }
+  last_buttonpress = is_button_pressed; 
+}
+
+///////////////////////////////////////
+// 4. Trunk charge animation
+//// Upon press, "charge up" the LEDs on the tree by turning on LEDs in series until reaching full bar
+///////////////////////////////////////
+void trunk_charge(){
+  // Charge up the trunk with a green light
+  if(is_button_pressed){
+  leds[pos] = CHSV(96, 237, 255);
+  pos++;
+  }
+  else {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(0, 0, 0);
+      pos = 0;
+      } //If no button press, LEDs off
+   }
+  FastLED.show();
+}
+
+///////////////////////////////////////
+// 5. Vibe pulse
+//// Upon press, pulse the LEDs on the recieved vibe - increase and decrease the speed of the pulse
+///////////////////////////////////////
+void vibe_pulse(){
+   if(is_button_pressed) {
+      for(int i = 0; i < NUM_LEDS; i++){
+		leds[i] = CHSV(global_hue, 200, bri);
+  }
+	}
+    else {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(0, 0, 0);
+      } //If no button press, LEDs off
+    }
+    FastLED.show(); 
+
+  if(is_button_pressed == true & last_buttonpress == true) {
+   bri += dir; // Increase brightness by 1
+  }
+  // Check if brightness maxed
+  if (bri >= 255) {
+    dir = -5; // decrease brightness
+  } else if (bri < 10) {
+    dir = 5; // increase brightenss
+  }
+  //Color change if button is released
+  if(is_button_pressed == false & last_buttonpress == true) {
+    global_hue = random8();
+    Serial.println(global_hue);
+  }
+  last_buttonpress = is_button_pressed;
+}
+
+///////////////////////////////////////
+// 6. Vibe Fireworks
+//// Upon press, flash hoops and bars at random times
+///////////////////////////////////////
+void vibe_fireworks(){
+
+static bool flashing = false;
+
+///green light to launch firework if probability condition is satisfied
+    if( random8() < chance_of_firework) {
+    go_firework = true;
+  }
+
+//If button is pressed for first time when animation is not flashing, then turn flashing on and log start of flashing animation
+if(is_button_pressed){
+  if(!flashing){
+    flashing_start_time = millis();
+    flashing = true;
+  }
+}
+// Flash firework if button is down and have green light to launch. Hold the flash for 1 second 
+    if(is_button_pressed && go_firework) {
+      if(millis() - flashing_start_time <= 1000){
+        for(int i = 0; i < NUM_LEDS; i++){
+		    uint8_t bri = 255;
+		    leds[i] = CHSV(global_hue, 200, bri);
+      }
+	}
+  //Fade out the firework after being shown for one second
+    else if(millis() - flashing_start_time >= 1000 && millis() - flashing_start_time <= 2000) {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(global_hue, 200, bri);
+      }
+      bri = bri - 5;
```

Comment:
Careful these are using two different values of `bri`.

line 265 is using the variable defined on 264.
line 271 is using the variable defined on line 49.